natss/pkg/reconciler/dispatcher/natsschannel_test.go

```diff
@@ -0,0 +1,202 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	fakekubeclientset "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/rest"
+	clientgotesting "k8s.io/client-go/testing"
+	"knative.dev/pkg/controller"
+	logtesting "knative.dev/pkg/logging/testing"
+	. "knative.dev/pkg/reconciler/testing"
+
+	clientset "knative.dev/eventing-contrib/natss/pkg/client/clientset/versioned"
+	informers "knative.dev/eventing-contrib/natss/pkg/client/informers/externalversions"
+	dispatchertesting "knative.dev/eventing-contrib/natss/pkg/dispatcher/testing"
+	"knative.dev/eventing-contrib/natss/pkg/reconciler"
+	reconciletesting "knative.dev/eventing-contrib/natss/pkg/reconciler/testing"
+)
+
+const (
+	testNS = "test-namespace"
+	ncName = "test-nc"
+)
+
+func TestAllCases(t *testing.T) {
+	ncKey := testNS + "/" + ncName
+
+	table := TableTest{
+		{
+			Name: "bad workqueue key",
```

Comment:
Instead of these comments, have you considered using more descriptive names such as "fails when the Channel name is invalid"?

Reply:
Good point! I see them copied all over the place in knative eventing. I will follow your idea here